src/Model/Casts/DateTime/DateTimeCast.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace Mongolid\Model\Casts\DateTime;
+
+use DateTime;
+use MongoDB\BSON\UTCDateTime;
+use Mongolid\Util\LocalDateTime;
+
+class DateTimeCast extends BaseDateTimeCast
+{
+    /**
+     * @param UTCDateTime|null $value
+     */
+    public function get(mixed $value): DateTime|null
```

Comment:
Mesma coisa aqui. Não rola simplificar com interrogação?
```suggestion
     * @param ?UTCDateTime $value
     */
    public function get(mixed $value): ?DateTime
```
E o mesmo se aplica pra todos outros casos de tipagem.

Reply:
https://github.com/leroy-merlin-br/mongolid/pull/220#discussion_r1367294696